### ISSUE TITLE
Raise PT011 only if exception info is unused

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#5157]
Make a new variant of `PT011` ([`pytest-raises-too-broad`](https://beta.ruff.rs/docs/rules/#flake8-pytest-style-pt)) that only triggers if the context is unused.

```py
# bad
with pytest.raises(ValueError):
    raise ValueError(f"unknown {option=}")

# ok
with pytest.raises(ValueError) as exc:
    raise ValueError(f"unknown {option=}")

assert f"unknown {option=}" in str(exc.value)
```

`PT011` currently triggers for both these cases even though the 2nd is plenty specific enough.

**Comment by irm-codebase** [comment]
Was about to open an issue on this, but it  seems this and  #6840 already discuss this.
I agree with  the suggestion.

Just trust us that the excinfo will be checked!

Here is my example (quite similar to #5157):

```python
    @pytest.mark.parametrize("mapping", [{}, {"foo": 1, "bar": 1}])
    def test_get_val_at_index_not_one_mapping(
        self, expression_get_val_at_index, mapping
    ):
        with pytest.raises(ValueError) as excinfo:  # noqa: PT011, false positive.
            expression_get_val_at_index(**mapping)
        assert check_error_or_warning(
            excinfo, "Supply one (and only one) dimension:index mapping"
        )
```
